src/NexusMods.App.UI/RightContent/Downloads/InProgressViewModel.cs

```diff
@@ -1,6 +1,89 @@
-﻿namespace NexusMods.App.UI.RightContent.Downloads;
+﻿using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using DynamicData;
+using NexusMods.App.UI.Overlays;
+using NexusMods.App.UI.Overlays.Download.Cancel;
+using NexusMods.App.UI.RightContent.Downloads.ViewModels;
+using NexusMods.App.UI.Windows;
+using NexusMods.Networking.Downloaders;
+using NexusMods.Networking.Downloaders.Interfaces;
+using ReactiveUI;
 
-public class InProgressViewModel : AViewModel<IInProgressViewModel>, IInProgressViewModel
+namespace NexusMods.App.UI.RightContent.Downloads;
+
+public class InProgressViewModel : InProgressCommonViewModel
 {
-    
+    public InProgressViewModel(DownloadService downloadService, IOverlayController overlayController)
+    {
+        SourceCache<IDownloadTaskViewModel, IDownloadTask> tasks = new(_ => throw new NotImplementedException());
+
+        // Add already running tasks to the tasks list
+        foreach (var task in downloadService.Downloads)
+        {
+            if (task.Status != DownloadTaskStatus.Completed)
+            {
+                tasks.Edit(x =>
+                {
+                    x.AddOrUpdate(new DownloadTaskViewModel(task), task);
+                });
+            }
+        }
+        
+        this.WhenActivated(d =>
+        {
+            ShowCancelDialog = ReactiveCommand.Create(async () =>
+            {
+                if (SelectedTask == null)
+                    return;
+                
+                var result = await overlayController.ShowCancelDownloadOverlay(SelectedTask);
+                if (result) 
+                    CancelSelectedTask();
+            });
+
+            // Subscribe to started tasks
+            downloadService.StartedTasks
```

Comment:
This will probably be much simplified with the switch to `ReadOnlyObservableCollection`